File: LIB_Common/src/main/java/com/xm/lib/common/MainActivity.java
package com.xm.lib.common;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.xm.lib.common.base.rv.v2.BaseRvAdapterV2;
import com.xm.lib.common.base.rv.v2.BaseViewHolderV2;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

import kotlin.Pair;

public class MainActivity extends AppCompatActivity {



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: app/src/main/java/com/xm/lib/test/ui/act/MainActivity.kt
package com.xm.lib.test.ui.act

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import com.xm.lib.common.base.rv.v2.BaseRvAdapterV2
import com.xm.lib.test.R
import com.xm.lib.test.holder.ItemVH

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val rv = findViewById<RecyclerView>(R.id.rv)
        rv.adapter = BaseRvAdapterV2.Builder()
                .addDataResouce(getDataResouce())
                .addHolderFactory(ItemVH.Factory())
                .build()
        rv.layoutManager = LinearLayoutManager(this)
        rv?.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
    }

    private fun getDataResouce(): ArrayList<Any> {
        applicationContext
        baseContext
        val items = ArrayList<Any>()
        items.add("Common")
        items.add("Component")
        items.add("Downloader")
        items.add("Media")
        items.add("Pay")
        items.add("Share")
        items.add("Statistics")
        items.add("Web")
        items.add("PrimaryTest")
        return items
    }
}
